src/modules/link-tooltip.coffee

```diff
@@ -100,8 +100,17 @@ class LinkTooltip extends Tooltip
 
   _expandRange: (range) ->
     [leaf, offset] = @quill.editor.doc.findLeafAt(range.start, true)
+
+    # Get all adjacent leaves from clicked leaf that share the same link format
+    leaves = [leaf]
+    if leaf?
```

Comment:
Is `leaf` ever null/falsy here? The previous code didn't do any checking before doing `end = start + leaf.length`...

Reply:
I'm pretty sure this is residual logic from a first-pass approach where `leaf` could be `null`. Will refactor out...